docs/building-a-form/creating-a-form-config-file.md

```diff
@@ -0,0 +1,273 @@
+              }
+            }
+          }
+        }
+      },
+
+      // See "About the `schema` and `uiSchema` objects" below.
+      uiSchema: {
+        'ui:title': 'My form',
+        field1: {
+          'ui:title': 'My field'
+        }
+      }
+    }
+  }
+}
+```
+
+### About the `schema` and `uiSchema` objects
+
+`uiSchema` is the object that provides information about how the page should be rendered. This object follows the format described in the [react-jsonschema-form documentation](https://github.com/mozilla-services/react-jsonschema-form#the-uischema-object), with some vets.gov-specific additions. The `schema` and `uiSchema` objects should have a similar structure, with the same fields organized in the same way with these exceptions:
+
+- `uiSchema` doesn't need to contain all the fields found in the `schema` object.
+- `uiSchema` doesn't need a `properties` object for sub-fields.
+
+For example, given this schema:
+
+```js
+{
+  type: 'object',
+  properties: {
+    field1: {
+      type: 'string'
+    }
+  }
+}
+```
+
+The matching `uiSchema` would be:
+
+```js
+{
+  'ui:title': 'My form',
+  field1: {
+    'ui:title': 'My field'
+  }
+}
+```
+
+For array fields, you must specify an `items` object that contains the fields for each row in the array in the `uiSchema` object:
+
+```js
+{
+  'ui:title': 'My form',
+  toursOfDuty: {
+    items: {
+      branchName: {
+        'ui:title': 'Branch'
+      }
+    }
+  }
+}
+```
+
+### Configuring `uiSchema` using rjsf options
+
+If you're not already familiar with the rjsf uiSchema options, see the [library docs](https://github.com/mozilla-services/react-jsonschema-form#the-uischema-object). Some commonly used options include:
+
+- [ui:order](https://github.com/mozilla-services/react-jsonschema-form#object-fields-ordering): An array of field names in the order in which they should appear.
+- [ui:widget](https://github.com/mozilla-services/react-jsonschema-form#alternative-widgets): The name of an alternative widget to use for the field, for example, a custom widget called `yesNo`.
+- [ui:field](https://github.com/mozilla-services/react-jsonschema-form#custom-field-components): The name of a custom field.
+- [classNames](https://github.com/mozilla-services/react-jsonschema-form#custom-css-class-names): The class names to put on the component.
+
+### Configuring `uiSchema` using US Form System options
+
+The us-forms-system code includes additional `uiSchema` functionality not found in the rjsf library.
+
+```js
+{
+  // Used instead of the `title` property in the JSON Schema.
+  'ui:title': '',
+  // It can also be a component, which passes the current form data as a property.
+  'ui:title': ({ formData }) => <legend>{`A ${formData.thing} title`}</legend>,
+
+  // Used instead of the `description` property in the JSON Schema. This can be a string or a React component, and is normally used on object fields in the schema to provide description text or HTML before a block of fields.
```

Comment:
Can we break these long comments into shorter multi-line comments? In a `js` code block they're rendering as one long line which means you have to scroll horizontally. (This applies to several code blocks in the docs but in particular this one and the one below.

Reply:
Do you have "Soft Wrap" or "Soft Wrap at Preferred Line Length" enabled? I'm not a fan of introducing line breaks into prose.

Reply:
I meant in the rendered Markdown, you can't see most of these long comments.

Reply:
BAH, you're right. 😩 Will fix.